app/models/lookup_value.rb

```diff
@@ -29,6 +29,13 @@ class LookupValue < ApplicationRecord
   scoped_search :on => :match, :complete_value => true
   scoped_search :relation => :lookup_key, :on => :key, :rename => :lookup_key, :complete_value => true
 
+  def create_or_update(*)
+    super
+  rescue ActiveRecord::RecordNotUnique
+    errors.add(:match, :taken)
```

Comment:
Did you handle this exception because an existing uniqueness validation defined using `:scope` & added index on :match with unique=true?

Reply:
this should be adding the same arror as the validation, but through the database error (when the validation does not catch such a case).

Reply:
I've replaced this with https://github.com/rails/rails/issues/20676#issuecomment-529110128 to take care of the uniqueness in validation phase ^_^ looks bit cleaner now.

Reply:
Cleaner plus efficient one :)